server_config_test.cc

```diff
@@ -20,130 +21,243 @@ class ServerConfigTest : public ::testing::Test {
 
+TEST_F(ServerConfigTest, BuildOK) {
+    std::unique_ptr<ServerConfig> server_config
+      = ParseString("port 8080; \
+                     path /echo EchoHandler {}");
+    bool ec = server_config->Build();
+
+    EXPECT_EQ(true, ec)
+      << "Build() should return true if a valid config is supplied to a \
+          Server Config is supplied";
+}
+
+TEST_F(ServerConfigTest, BuildFailed) {
+    std::unique_ptr<ServerConfig> server_config = ParseString("");
+    bool ec = server_config->Build();
+
+    EXPECT_EQ(false, ec)
+      << "Build() should return false if an empty config is supplied";
+}
+
+TEST_F(ServerConfigTest, AllPathsNotEmpty) {
+    std::unique_ptr<ServerConfig> server_config
+      = ParseString("port 8080; \
+                     path /echo EchoHandler {} \
+                     default NotFoundHandler {}");
+    bool ec = server_config->Build();
+    std::unordered_map<std::string, std::string> paths = server_config->allPaths();
+
+    EXPECT_EQ(true, ec)
+      << "Build() should return true if a valid config is supplied";
+    EXPECT_EQ(false, paths.empty())
+      << "allPaths() shouldn't return an empty map";
+}
+
+TEST_F(ServerConfigTest, AllPathsEmpty) {
+    std::unique_ptr<ServerConfig> server_config = ParseString("port 8080;");
+    bool ec = server_config->Build();
+    std::unordered_map<std::string, std::string> paths = server_config->allPaths();
+
+    EXPECT_EQ(true, ec)
+      << "Build() should return true if a valid config is supplied";
+    EXPECT_EQ(true, paths.empty())
+      << "allPaths() should return an empty map";
+}
+
+TEST_F(ServerConfigTest, HasChildBlock) {
+    std::unique_ptr<ServerConfig> server_config
+      = ParseString("port 8080; \
+                     path / StaticHandler { \
+                       root /foo/bar \
+                     }");
+    bool ec = server_config->Build();
+    std::unique_ptr<NginxConfig> childBlock = server_config->getChildBlock("/");
+
+    EXPECT_EQ(true, ec)
+      << "Build() should return true if a valid config is supplied";
+    EXPECT_EQ(1, childBlock->statements_.size())
+      << "The child block of '/' prefix should have one statement in it";
+}
+
+TEST_F(ServerConfigTest, EmptyChildBlock) {
+    std::unique_ptr<ServerConfig> server_config
+      = ParseString("port 8080; \
+                     path / StaticHandler {}");
+    bool ec = server_config->Build();
+    std::unique_ptr<NginxConfig> childBlock = server_config->getChildBlock("/");
+
+    EXPECT_EQ(true, ec)
+      << "Build() should return true if a valid config is supplied";
+    EXPECT_EQ(0, childBlock->statements_.size())
+      << "The child block of '/' prefix should have no statements";
+}
+
 TEST_F(ServerConfigTest, NoPortConfig) {
     std::unique_ptr<ServerConfig> server_config = ParseString("server { ACM Hack; }");
-    std::vector<std::string> query = {"server", "listen"};
+    bool buildCode = server_config->Build();
+
+    std::vector<std::string> query = {"server", "port"};
```

Comment:
Could we place the test code + accompanying assertions together? That way, if we have more assertions later on, we don't have to scan up-and-down to pair up test-case with assertion.

For example: 

```c++
const auto actual_result1 = testCase1();
EXPECT_EQ(expected, actual_result);

const auto actual_result2 = testCase2();
EXPECT_EQ(expected, actual_result);
```

As compared to: 

```c++
const auto actual_result1 = testCase1();
const auto actual_result2 = testCase2();

EXPECT_EQ(expected, actual_result1);
EXPECT_EQ(expected, actual_result2);
```